torch/nn/functional.py

```diff
@@ -2798,46 +2786,45 @@ def gaussian_nll_loss(
             eps=eps,
             reduction=reduction,
         )
-
-    # Check var size
-    # If var.size == input.size, the case is heteroscedastic and no further checks are needed.
-    # Otherwise:
-    if var.size() != input.size():
-
-        # If var is one dimension short of input, but the sizes match otherwise, then this is a homoscedastic case.
-        # e.g. input.size = (10, 2, 3), var.size = (10, 2)
-        # -> unsqueeze var so that var.shape = (10, 2, 1)
-        # this is done so that broadcasting can happen in the loss calculation
-        if input.size()[:-1] == var.size():
+    # Check var shape
+    # var, which is a tensor of variances, can either represent a homoscedastic
+    # assumption (data has the same variance) or a heteroscedastic one (data has
+    # different variances).
+    #
+    # If var is the same shape as input, it's the heteroscedastic case.
+    # If var is *not* the same shape as input, it's the homoscedastic case.
+    #
+    # To support broadcasting, the following sub-cases are allowed in the
```

Comment:
It's not really broadcasting, though. We should be clear about what, exactly, it does, and not use a similar concept, which could confuse the reader